From 5c1c4e0c348d575cdb80ca2ddd82087009c34af8 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 26 Jan 2024 18:03:58 +0000
Subject: [PATCH 1/1] Install icon to standard path, fix appstream-builder
 generation

appstream-builder was failing to generate the appstream metainfo from
this package as it didn't provide an icon in a standard path.

appstream-builder states:
> Icons MUST be installed in /usr/share/pixmaps/*, /usr/share/icons/*,
> /usr/share/icons/hicolor/*/apps/*, or /usr/share/${app_name}/icons/*
---
 data/input-remapper-autoload.desktop | 2 +-
 data/input-remapper-gtk.desktop      | 2 +-
 setup.py                             | 1 +
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/data/input-remapper-autoload.desktop b/data/input-remapper-autoload.desktop
index de6ee98..a647840 100644
--- a/data/input-remapper-autoload.desktop
+++ b/data/input-remapper-autoload.desktop
@@ -2,5 +2,5 @@
 Type=Application
 Exec=bash -c "input-remapper-control --command stop-all && input-remapper-control --command autoload"
 Name=input-remapper-autoload
-Icon=/usr/share/input-remapper/input-remapper.svg
+Icon=input-remapper
 Comment=Starts injecting all presets that are set to automatically load for the user
diff --git a/data/input-remapper-gtk.desktop b/data/input-remapper-gtk.desktop
index 3255e6a..71785a3 100644
--- a/data/input-remapper-gtk.desktop
+++ b/data/input-remapper-gtk.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=Input Remapper
-Icon=/usr/share/input-remapper/input-remapper.svg
+Icon=input-remapper
 Exec=input-remapper-gtk
 Terminal=false
 Categories=Settings
diff --git a/setup.py b/setup.py
index 20c7d3c..5b768ae 100644
--- a/setup.py
+++ b/setup.py
@@ -119,6 +119,7 @@ setup(
             "/usr/share/metainfo/",
             ["data/io.github.sezanzeb.input_remapper.metainfo.xml"],
         ),
+        ("/usr/share/icons/hicolor/scalable/apps/", ["data/input-remapper.svg"]),
         ("/usr/share/polkit-1/actions/", ["data/input-remapper.policy"]),
         ("/usr/lib/systemd/system", ["data/input-remapper.service"]),
         ("/etc/dbus-1/system.d/", ["data/inputremapper.Control.conf"]),
-- 
2.43.0

